src/app/modules/shared-components/components/switch/switch.component.ts

```diff
@@ -17,8 +21,8 @@ export class SwitchComponent implements OnInit {
 
   @Output() change = new EventEmitter<boolean>();
 
-  ngOnInit(): void {
-    if (this.parentForm && this.parentForm.get(this.name)) {
+  ngOnChanges(changes: SimpleChanges): void {
+    if (Object.prototype.hasOwnProperty.call(changes, 'parentForm') && this.parentForm && this.parentForm.get(this.name)) {
```

Comment:
The `name` could be changed as well technically, as it is an input.

Reply:
fixed